packages/ui/src/overview/components/DeadlineList/ElectionListItem.tsx

```diff
@@ -22,15 +27,19 @@ import {
 } from './styles'
 
 export interface ElectionListItemProps {
-  electionId: string
-  title: string
+  election: Election
   hideForStorage: (id: string) => void
+  member: Member
 }
 
-export const ElectionListItem: React.FC<ElectionListItemProps> = React.memo(({ electionId, title, hideForStorage }) => {
+export const ElectionListItem: React.FC<ElectionListItemProps> = React.memo(({ hideForStorage, member, election }) => {
   const { t } = useTranslation('overview')
   const { stage: electionStage } = useElectionStage()
+  const { allAccounts } = useMyAccounts()
+  const { votesTotal } = useMyCurrentVotesCount(election.cycleId)
+  const canVote = isDefined(votesTotal) && allAccounts.length > votesTotal
   const remainingPeriod = useElectionRemainingPeriod(electionStage)
+  const { votes } = useMyCastVotes(election.cycleId)
```

Comment:
We can skip this hook if we're not on the `revealing` stage and also it would simplify the logic to default to an empty array

```suggestion
  const { votes } = useMyCastVotes(electionStage === 'revealing' ? election.cycleId : undefined) ?? []
```